11-payment-encoding.md

```diff
@@ -0,0 +1,339 @@
+node ID of the start of the channel, `channel_id` is the channel ID
+field to identify the channel, `fee` is the total fee required to use
+that channel to send `amount` to the final node, specified in 10^-11
+currency units, and `cltv_expiry_delta` is the block delta required
+by the channel.  A writer MAY include more than one `r` field to
+indicate a sequence of non-public channels to traverse.
+
+A writer MUST pad fields to a multiple of 5 bits, using zeroes.
+
+A reader MUST skip over unknown fields, or `f` fields which do not
+match the descriptions here.  A reader MAY fail or ignore a known
+field which has an unexpected length.  A reader MUST check that the
+SHA-2 256 in the `h` field exactly matches the hashed description.
+
+### Rationale
+
+The type-and-length format allows future extensions to be backward
+compatible.  Lengths are always a multiple of 5 bits, for easy
+encoding and decoding.
+
+The `d` field allows inline descriptions, but may be insufficient for
+complex orders; thus the `h` field allows a summary, though the method
+by which the description is served is as-yet unspecified, and will
+probably be transport-dependent.
+
+The `x` field gives advance warning as to when a payment will be
+refused; this is mainly to avoid confusion.  The default was chosen
+to be reasonable for most payments, and allow sufficient time for
+on-chain payment if necessary.
+
+The `f` field allows on-chain fallback.  This may not make sense for
+tiny or very time-sensitive payments, however.  It's possible that new
+address forms will appear, and so multiple `f` fields in an implied
+preferred order help with transition.
+
+The `r` field allows limited routing assistance: as specified it only
+allows minimum information to use private fields, but it could also
+assist in future partial-knowledge routing.
+
+# Payer / Payee Interactions
+
+These are generally defined by the rest of the lightning BOLT series,
+but it's worth noting that BOLT #5 specifies that the payee SHOULD
+accept up to twice the expected `amount`, so the payer can make
+payments harder to track by adding small variations.
+
+The intent is that the payer recover the payee's node ID from the
+signature, and after checking the conditions are acceptable (fees,
+expiry, block timeout), attempt a payment.  It can use `r` fields to
+augment its routing information if necessary to reach the final node.
+
+If the payment succeeds but there is a later dispute, the payer can
+prove both the signed offer from the payee, and the successful
+payment.
+
+## Payer / Payee Requirements
+
+A payer SHOULD NOT attempt a payment after the `timestamp` plus
+`expiry` has passed.  Otherwise, if a lightning payment fails, a payer
+MAY attempt to use the address given the first `f` field it
+understands for payment.  A payer MAY use the sequence of channels
+specified by `r` to route to the payee.  A payee SHOULD consider the
+fee amount and payment timeout before initiating payment.
```

Comment:
Unless I'm confused, you meant "payer SHOULD consider the fee amount and payment timeout before initiating payment".